sections/errorhandling/failfast.md

```diff
@@ -37,8 +62,24 @@ function redirectToPrintDiscount(httpResponse, member, discount) {
 
 redirectToPrintDiscount(httpResponse, someMember);
 // forgot to pass the parameter discount, why the heck was the user redirected to the discount screen?
+```
+</details>
+
+<details>
+<summary><strong>Typescript</strong></summary>
+
+```typescript
+// if the discount is positive let's then redirect the user to print his discount coupons
+function redirectToPrintDiscount(httpResponse: Response, member: Member, discount: number) {
+  if (discount != 0) {

```

Comment:
I'm not sure about this example both in JavaScript and TypeScript since it's rather about language features than validation in this case, right? We don't have to remove it completely but in general the solution would rather be to use a different equality operator (`!==`) in this case to avoid implicit type conversion from the undefined value coming from the unsupplied discount and rather only check on the numerical value

Reply:
That's true, maybe the example does not perfectly suit the point we want to illustrate here. 

What come to my mind when I think about this problem is what often happens when we receive a big json object with one missing value and forgot to check it. It goes all the way to the DB to get a constraint error (or worse, if there's not constraint). That could be avoided with schema validation. But that's harder to illustrate. Any other idea?

Reply:
Yeah that's definitely the right use case for validation, I think the "no validation yields bugs" example can be visualized with various cases, for example sending an email with some helper function and receiving an invalid value causing a crash, these things will always need to be validated, I guess.

Reply:
Thinking a bit more about this, shouldn't this be another PR?
It's more about the existing example not accurately describing the problem than the translation of it in Typescript. I feel like that should be discussed in a separate issue.

Reply:
@kevynb Yes, sounds like another PR but it's a good point that you and @BrunoScheufler raised